sql-plugin/src/main/scala/com/nvidia/spark/rapids/DumpUtils.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2020-2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids
+
+import java.io.File
+import java.text.SimpleDateFormat
+import java.util.{Date, Locale, Random}
+
+import ai.rapids.cudf.CompressionType
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.mapred.JobID
+import org.apache.hadoop.mapreduce.{TaskAttemptContext, TaskAttemptID, TaskID, TaskType}
+import org.apache.hadoop.mapreduce.task.TaskAttemptContextImpl
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.execution.datasources.parquet.ParquetWriteSupport
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.internal.SQLConf.ParquetOutputTimestampType
+import org.apache.spark.sql.types.{StructField, StructType}
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+object DumpUtils extends Logging {
+  /**
+   * Debug utility to dump columnar batch to parquet file. <br>
+   * It's running on GPU. Parquet column names are generated from columnar batch type info
+   * @param columnarBatch the columnar batch to be dumped, should be GPU columnar batch
+   * @param filePrefix parquet file prefix, e.g. /tmp/my-debug-prefix-
+   * @return parquet file path if dump is successful, e.g. /tmp/my-debug-prefix-123.parquet
+   */
+  def dumpToParquetFile(columnarBatch: ColumnarBatch, filePrefix: String): Option[String] = {
+    try {
+      if(columnarBatch.numCols() == 0) {
+        logWarning("dump to parquet failed, has no column, file prefix is " + filePrefix)
+        None
+      } else {
+        Some(dumpToParquetFileImpl(columnarBatch, filePrefix))
+      }
+    } catch {
+      case e: Exception =>
```

Comment:
Why is this suppressing exceptions?  Do we think it's common that we wouldn't mind if the data fails to write?  I'd rather have the caller suppress exceptions when desired if that's not the common case.

Reply:
done, will throw if exception occurs